src/legacy/core_plugins/metrics/public/components/vis_editor.js

```diff
@@ -52,23 +55,45 @@ class VisEditor extends Component {
     return this.props.config.get(...args);
   };
 
-  handleUiState = (field, value) =>  {
+  handleUiState = (field, value) => {
     this.props.vis.uiStateVal(field, value);
   };
 
   handleChange = async (partialModel) => {
-    const nextModel = { ...this.state.model, ...partialModel };
+    if (isEmpty(partialModel)) {
+      return;
+    }
+    const hasTypeChanged = partialModel.type && this.state.model.type !== partialModel.type;
+    const nextModel = {
+      ...this.state.model,
+      ...partialModel,
+    };
+    let dirty = true;
+
     this.props.vis.params = nextModel;
-    if (this.state.autoApply) {
+
+    if (this.state.autoApply || hasTypeChanged) {
       this.props.vis.updateState();
+
+      dirty = false;
     }
+
+    if (this.props.isEditorMode) {
+      const { params, fields } = this.props.vis;
+      const extractedIndexPatterns = extractIndexPatterns(params, fields);
+
+      if (!isEqual(this.state.extractedIndexPatterns, extractedIndexPatterns)) {
```

Comment:
nit: we can avoid fetching indexes fields also when these two arrays has the same elements but in a different order (like reordering the series).

Reply:
@markov00  Good point!  Fixed in `extractIndexPatterns `method. Now this metrhod returns sorted array